From ad4a871df4ae458f1b28a8b08790be5a5eb6cfe3 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 01 Aug 2021 12:41:44 +0300
Subject: [PATCH] [SW-51308] gaudi: security: set pci gw prot secure

The PCIe GW initiator should have protection level set to secured.
Make sure it is so.

Change-Id: I308af844317f0e11812ea12229938978af677933
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index e0f46af..be18e5e 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -836,6 +836,10 @@
 		PCIE_WRAP_LBW_PROT_OVR_RD_EN_MASK;
 	regmap_writel(mask, mmPCIE_WRAP_LBW_PROT_OVR);
 
+	/* Set PCI GW level to secured */
+	mask = PROT_OVR_SECURE << PCIE_WRAP_IND_AWPROT_VAL_SHIFT;
+	regmap_writel(mask, mmPCIE_WRAP_IND_AWPROT);
+
 	prot_base = base;	/* offsets: 0 - 0x200 */
 	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 4);
 
